### ISSUE TITLE
synchronize pypi version with the tag 1.10.0

### ISSUE THREAD

**Issue by fblackburn1** [hamcrest/PyHamcrest#130]
Do you plan to synchronize the pypi version with the `1.10.0` tag?
Currently, the pypi version is synced to the `release-1.10 `branch but do not contains some quick fix that is in the `1.10.0` tag

Quick fix: https://github.com/hamcrest/PyHamcrest/commit/b68b70d7729529d477c052f208a2fae9222e0d21#diff-d8329dda1453e5b8c0430eb4d945610eR34

Now, with the pypi version, I get a lot of
```
  File "/home/francois/.virtualenvs/dao/lib/python3.7/site-packages/hamcrest/core/assert_that.py", line 45, in assert_that
    _assert_match(actual=arg1, matcher=arg2, reason=arg3)
  File "/home/francois/.virtualenvs/dao/lib/python3.7/site-packages/hamcrest/core/assert_that.py", line 53, in _assert_match
    if not matcher.matches(actual):
  File "/home/francois/.virtualenvs/dao/lib/python3.7/site-packages/hamcrest/core/core/described_as.py", line 20, in matches
    return self.matcher.matches(item, mismatch_description)
  File "/home/francois/.virtualenvs/dao/lib/python3.7/site-packages/hamcrest/core/core/allof.py", line 20, in matches
    if not matcher.matches(item):
  File "/home/francois/.virtualenvs/dao/lib/python3.7/site-packages/hamcrest/core/base_matcher.py", line 27, in matches
    match_result = self._matches(item)
  File "/home/francois/.virtualenvs/dao/lib/python3.7/site-packages/hamcrest/library/object/hasproperty.py", line 26, in _matches
    return self.value_matcher.matches(value)
  File "/home/francois/.virtualenvs/dao/lib/python3.7/site-packages/hamcrest/library/collection/issequence_containinginanyorder.py", line 54, in matches
    if not matchsequence.matches(item):
  File "/home/francois/.virtualenvs/dao/lib/python3.7/site-packages/hamcrest/library/collection/issequence_containinginanyorder.py", line 16, in matches
    return self.isnotsurplus(item) and self.ismatched(item)
  File "/home/francois/.virtualenvs/dao/lib/python3.7/site-packages/hamcrest/library/collection/issequence_containinginanyorder.py", line 36, in ismatched
    if matcher.matches(item):
  File "/home/francois/.virtualenvs/dao/lib/python3.7/site-packages/hamcrest/core/core/described_as.py", line 20, in matches
    return self.matcher.matches(item, mismatch_description)
  File "/home/francois/.virtualenvs/dao/lib/python3.7/site-packages/hamcrest/core/core/allof.py", line 29, in matches
    mismatch_description.append_text(" and ")
AttributeError: 'NoneType' object has no attribute 'append_text'
```

**Comment by brunns** [comment]
Hmmm. I'm confused. The Pypi version should contain exactly what's in the tag.

**Comment by brunns** [comment]
Looks like the tag is wrong - it contains backward incompatible changes which are intended for 2.0, but not for 1.10. I don't know how that happened.

**Comment by brunns** [comment]
The tag and release should match now. Are you still having problems?

**Comment by fblackburn1** [comment]
> The tag and release should match now. Are you still having problems?

Yes now the pypi release and tag `V1.10.0` match.
But the tag `V1.10.0` and `1.10.0` are on different commit
It doesn't solve my initial problem, but since the fix I want is in master, I just wait for the 2.0.0 or 1.10.1 and pin all my requirements to 1.9.0 for now.

Thanks

**Comment by brunns** [comment]
Which fix exactly is missing from 1.19.0?

**Comment by fblackburn1** [comment]
> Which fix exactly is missing from 1.19.0?

Is not a fix missing from 1.9.0, it is a regression in 1.10.0 from 1.9.0.

With version 1.9.0 you can call [AllOf without `mismatch_description`](https://github.com/hamcrest/PyHamcrest/blob/V1.9.0/src/hamcrest/core/core/allof.py#L14) and all is fine
Now with 1.10.0, if you call AllOf without mismatch_description, the default value of None can raise `AttributeError: 'NoneType' object has no attribute 'append_text'` on [this line](https://github.com/hamcrest/PyHamcrest/blob/V1.10.0/src/hamcrest/core/core/allof.py#L29)

This behavior has been fixed in master only with the following commit: https://github.com/hamcrest/PyHamcrest/commit/b68b70d7729529d477c052f208a2fae9222e0d21#diff-d8329dda1453e5b8c0430eb4d945610eR34

**Comment by brunns** [comment]
Ah, I see. That's a big old commit there, so I'll see if I can find the fix
and make a 1.10.1 release in the morning.
